### PR TITLE
OPTIM: faster SVD

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
Automatically switch between randomized SVD and lapack

### CODE REVIEW

**Comment by AlexandreAbraham** [comment]
Great! Do you have a bench to see the difference by any chance?

**Comment by GaelVaroquaux** [comment]
> Great! Do you have a bench to see the difference by any chance?

No. I trusted Giorgio's bench. But in the circleCI build it didn't make
any difference :(.

I think that we should still merge, as the code will be faster in certain
situations and it's rather more clean.

The test that I had written had a stupid loop, by the way. I fixed it.

**Comment by AlexandreAbraham** [comment]
> But in the circleCI build it didn't make any difference :(.

Not a surprise given that we try to keep the examples minimal. Maybe we can launch canica on a big dataset to see the difference?

**Comment by GaelVaroquaux** [comment]
> Not a surprise given that we try to keep the examples minimal.

Also, it's smoothing that is dominating this example.

> Maybe we can launch canica on a big dataset to see the difference?

I tried, but without a big subject-level data (eg HCP), the cost of the
SVDs is lost in things like filtering, smoothing, and the multiple ICAs.
:(

File: nilearn/decomposition/base.py [hunk 1]
```diff
@@ -4,22 +4,60 @@
...
 from .._utils.niimg import _safe_get_data
 from .._utils.compat import _basestring
 from ..input_data import NiftiMapsMasker
 from ..input_data.masker_validation import check_embedded_nifti_masker
 
 
+def fast_svd(X, n_components, random_state=None):
+    """ Automatically switch between randomized and lapack SVD (heuristic
+        of scikit-learn).
+    """
```
Comment (KamalakerDadi) [inline nilearn/decomposition/base.py]: I think that documentation could be useful and not supposed to a private function. Could you please add documentation ?

**Comment by GaelVaroquaux** [comment]
> I think that documentation could be useful and not supposed to a private function. Could you please add documentation ?

I added a docstring. But this is supposed to be a private function, hence
we don't want it in the narrative documentation.

**Comment by GaelVaroquaux** [comment]
Can I haz merge?

**Comment by GaelVaroquaux** [comment]
OK, merging, thanks!

**Pull request merged**